api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,222 @@
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      cpu:
+	//        min: "16"
+	//        max: "64"
+	//      memory:
+	//        min: "128Gi"
+	//        max: "512Gi"
+	//      accelerators:
+	//        nvidia.com/gpu:
+	//          min: "2"
+	//	        max: "8"
+	//  acceleratorSelector:
+	//    type: "nvidia.com/gpu"
+	//    memory: "48Gi"
+	//    count: 4
+	// +required
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// PerAcceleratorCount defines pre-configuration quantities for each accelerator type
+	PerAcceleratorCount []PerAcceleratorCount `json:"perAcceleratorCount"`
+}
+
+// PerAcceleratorCount defines configuration for specific accelerators
+type PerAcceleratorCount struct {
+	// AcceleratorSelector accelerator selector
+	AcceleratorSelector AcceleratorSelector `json:"acceleratorSelector"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node per accelerator.
+	// If two different counts are specified for the same (Node, Accelerator, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, Accelerator, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +kubebuilder:validation:MinItems=1
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirements defines resource requirements for a node.
+type ResourceRequirements struct {
+	// Allocatable defines the allocatable resources for a node with min/max ranges.
+	// +kubebuilder:validation:Required
+	Allocatable ResourceRanges `json:"allocatable,omitempty"`
+}
+
+// ResourceRanges defines min/max ranges for various resources of a Node.
+type ResourceRanges struct {
+	// CPU defines the CPU resource range requirement.
+	// +kubebuilder:validation:Optional
+	CPU ResourceRange `json:"cpu,omitempty"`
+
+	// Memory defines the memory resource range requirement.
+	// +kubebuilder:validation:Optional
+	Memory ResourceRange `json:"memory,omitempty"`
+
+	// Accelerators defines the GPU resource range requirements keyed by GPU type.
+	// +kubebuilder:validation:Optional
+	Accelerators map[string]ResourceRange `json:"accelerators,omitempty"`
+}
+
+// ResourceRange defines a range with minimum and maximum quantity values.
+type ResourceRange struct {
+	// Min specifies the minimum quantity required.
+	// +kubebuilder:validation:Optional
+	Min resource.Quantity `json:"min,omitempty"`
+
+	// Max specifies the maximum quantity allowed.
+	// +kubebuilder:validation:Optional
+	Max resource.Quantity `json:"max,omitempty"`
+}
+
+// EnhancedNodeSelector defines node selector with label selector and resource requirements.
+type EnhancedNodeSelector struct {
```

Comment:
This file is getting large and complicated enough that I think it should have a consistent and readily grasped ordering among its declarations. I suggest that declarations appear after first use, and in order of first use. That would make `EnhancedNodeSelector` come before `PerAcceleratorCount`.

Reply:
done